arangod/MMFiles/MMFilesWalRecoverState.cpp

```diff
@@ -444,6 +444,31 @@ bool MMFilesWalRecoverState::InitialScanMarker(MMFilesMarker const* marker, void
 bool MMFilesWalRecoverState::ReplayMarker(MMFilesMarker const* marker,
                                           void* data, MMFilesDatafile* datafile) {
   MMFilesWalRecoverState* state = reinterpret_cast<MMFilesWalRecoverState*>(data);
+  auto visitHelpers = std::shared_ptr<MMFilesWalRecoverState>(
+    state,
+    [marker](MMFilesWalRecoverState* state)->void {
```

Comment:
destructor is not marked as `noexcept` which may cause the whole process termination

Reply:
true